qiskit/ignis/verification/randomized_benchmarking/dihedral.py

```diff
@@ -559,6 +585,131 @@ def dot(self, other):
+
+        for i in range(elem0.num_qubits):
+            value = elem0.poly.get_term([i])
+            result.poly.set_term([i], value)
+            for j in range(i):
+                value = elem0.poly.get_term([j, i])
+                result.poly.set_term([j, i], value)
+                for k in range(j):
+                    value = elem0.poly.get_term([k, j, i])
+                    result.poly.set_term([k, j, i], value)
+
+        for i in range(elem1.num_qubits):
+            value = elem1.poly.get_term([i])
+            result.poly.set_term([i + elem0.num_qubits], value)
+            for j in range(i):
+                value = elem1.poly.get_term([j, i])
+                result.poly.set_term([j + elem0.num_qubits, i + elem0.num_qubits], value)
+                for k in range(j):
+                    value = elem1.poly.get_term([k, j, i])
+                    result.poly.set_term([k + elem0.num_qubits, j + elem0.num_qubits,
+                                          i + elem0.num_qubits], value)
+
+        return result
+
+    def tensor(self, other):
+        """Return the tensor product operator: self tensor other.
+
+         Args:
+             other (CNOTDihedral): an operator subclass object.
+         Returns:
+             CNOTDihedral: the tensor product operator: self tensor other.
+         """
+
+        return self._tensor_product(other, reverse=True)
+
+    def expand(self, other):
+        """Return the tensor product operator: other tensor self.
+
+         Args:
+             other (CNOTDihedral): an operator subclass object.
+         Returns:
+             CNOTDihedral: the tensor product operator: other tensor other.
+         """
+
+        return self._tensor_product(other, reverse=False)
+
+    def adjoint(self):
+        """Return the conjugate transpose of the CNOTDihedral element"""
+
+        circ = self.to_instruction()
+        result = self.from_circuit(circ.inverse())
+        return result
+
+    def conjugate(self):
+        """Return the conjugate of the CNOTDihedral element."""
+        circ = self.to_instruction()
+        new_circ = QuantumCircuit(self.num_qubits)
+        qargs = list(range(self.num_qubits))
+        for instr, qregs, _ in circ.definition:
+            new_qubits = [qargs[tup.index] for tup in qregs]
+            if instr.name == 'u1':
+                params = 2 * np.pi - instr.params[0]
+                instr.params[0] = params
+                new_circ.append(instr, new_qubits)
+            else:
+                new_circ.append(instr, new_qubits)
+        result = self.from_circuit(new_circ)
+        return result
+
+    def transpose(self):
+        """Return the transpose of the CNOT-Dihedral element."""
+
+        circ = self.to_instruction()
+        result = self.from_circuit(circ.reverse_ops())
+        return result
+
+    def is_cnotdihedral(self):
+        """Return True if input is a CNOTDihedral element."""
+
+        assert self.poly.weight_0 == 0
```

Comment:
Using `assert` will raise an exception. This should return False if any of the clauses fail, not raise an exception.